src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/RetryPolicy.cs

```diff
@@ -93,7 +97,30 @@ internal virtual async Task DelayAsync(TimeSpan time, CancellationToken cancella
             await Task.Delay(time, cancellationToken).ConfigureAwait(false);
         }
 
-        protected abstract void GetDelay(HttpPipelineMessage message, int attempted, out TimeSpan delay);
+        protected virtual void GetDelay(HttpPipelineMessage message, int attempted, out TimeSpan delay)
+        {
+            delay = TimeSpan.Zero;
+            if (message.Response.TryGetHeader(RetryAfterHeaderName, out var retryAfterValue))
```

Comment:
How did we decide the order of precedence for the headers?   I would agree with the chosen ordering, though I'm curious.   

Also, it is probably worth considering a comment to make it clear what happens if we see multiple retry headers, even though it's a `protected` method.

Reply:
The only scenario I can think of that would have both sets of headers is using `Retry-After` as a fallback for clients that don't support `retry-after-ms` or `x-ms-retry-after-ms`:

```
Retry-After: 1
x-ms-retry-after-ms: 200
```

In this case, we would want to pick the more precise value.

Reply:
Makes sense;  I wouldn't expect a well-behaved service to use more than a single header.   That said, I've also seen enough services to know that I don't expect them to be well behaved.   :)